vscode/src/azure/workspaceActions.ts

```diff
@@ -115,123 +122,175 @@ workspace = azure.quantum.Workspace(
   return pythonCode;
 }
 
-export async function queryWorkspaces(): Promise<
-  WorkspaceConnection | undefined
-> {
-  log.debug("Querying for account workspaces");
-  const associationId = getRandomGuid();
-  const start = performance.now();
-  sendTelemetryEvent(EventType.QueryWorkspacesStart, { associationId }, {});
-  // *** Authenticate and retrieve tenants the user has Azure resources for ***
+async function getWorkspaceWithConnectionString(
+  endEventProperties: EndEventProperties,
+): Promise<WorkspaceConnection | undefined> {
+  for (;;) {
+    const connStr = await vscode.window.showInputBox({
+      prompt: "Enter the connection string",
+      placeHolder:
+        "SubscriptionId=<guid>;ResourceGroupName=<name>;WorkspaceName=<name>;ApiKey=<secret>;QuantumEndpoint=<serviceUri>;",
+    });
+    if (!connStr) {
+      endEventProperties.reason = "no connection string entered";
+      endEventProperties.flowStatus = UserFlowStatus.Aborted;
+      return;
+    }
 
-  // For the MSA case, you need to query the tenants first and get the underlying AzureAD
-  // tenant for the 'guest' MSA. See https://stackoverflow.microsoft.com/a/76246/108570
-  const firstAuth = await getAuthSession([scopes.armMgmt], associationId);
+    const partsMap = new Map<string, string>();
+    connStr.split(";").forEach((part) => {
+      const eq = part.indexOf("=");
+      if (eq === -1) return;
+      partsMap.set(part.substring(0, eq).toLowerCase(), part.substring(eq + 1));
+    });
 
-  if (!firstAuth) {
-    log.error("No authentication session returned");
-    sendTelemetryEvent(
-      EventType.QueryWorkspacesEnd,
-      {
-        associationId,
-        reason: "no auth session returned",
-        flowStatus: UserFlowStatus.Failed,
-      },
-      { timeToCompleteMs: performance.now() - start },
-    );
-    return;
-  }
+    if (
+      !partsMap.has("subscriptionid") ||
+      !partsMap.has("resourcegroupname") ||
+      !partsMap.has("workspacename") ||
+      !partsMap.has("apikey") ||
+      !partsMap.has("quantumendpoint")
+    ) {
+      const action = await vscode.window.showErrorMessage(
+        "Invalid connection string. Please follow the placeholder format.",
```

Comment:
It could be multiple, or even all if it's just totally invalid nonsense. I don't think adding that much logic adds a lot more clarity beyond saying to follow the placeholder that is shown.